gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -590,6 +609,87 @@ def filter_freq_arrays(
+    """
+    Get the freq sample counts from the v3.1 subsets.
+
+    The freq sample counts on the v3.1.2 release sites HT only contains the counts for
+    the non subset groupings. This function adds the freq sample counts from the v3.1
+    subset frequency groups to the v3.1.2 sample count array to give a sample count
+    array matching the v3.1.2 'freq_meta' global annotation.
+
+    :param ht: Table of the 3.1.2 release sites HT, which only contains the freq
+       sample counts for non subset groupings.
+    :return: Table with the freq sample counts from the v3.1 subset groups added.
+    """
+    sc = ht.index_globals().freq_sample_count
+
+    for subset in SUBSETS:
+        freq_subset_ht = get_freq(
+            version="3.1", subset=subset, het_nonref_patch=True
+        ).ht()
+        sc_subset = freq_subset_ht.index_globals().freq_sample_count
+        sc = sc.extend(sc_subset)
+
+    ht = ht.annotate_globals(freq_meta_sample_count=sc)
+
+    return ht
+
+
+def update_pop_labels(ht: hl.Table, pop_map: Dict[str, Any]) -> hl.Table:
+    """
+    Update the population labels in the 'freq_meta' annotation on `ht`.
+
+    :param ht: Table with population labels to update.
+    :param pop_map: Dictionary mapping old to new population labels.
+    :return: Table with updated population labels in 'freq_meta' annotation.
+    """
+    pop_map = hl.literal(pop_map)
+    ht = ht.annotate_globals(
+        freq_meta=ht.freq_meta.map(
+            lambda d: d.map_values(lambda x: hl.or_else(pop_map.get(x), x))
+        )
+    )
+
+    return ht
+
+
+def join_release_ht_with_subsets(
+    ht: hl.Table, freq_hts: Dict[str, hl.Table]
+) -> hl.Table:
+    """
+    Join the release HT with the call stats for added and subtracted samples.
+
+    :param ht: gnomAD v3.1.2 release sites Table.
+    :param freq_hts: Dictionary with the call stats Tables for added and subtracted
+        samples.
+    :return: Table with `ht` and all Tables in `freq_hts` joined.
+    """
+    logger.info("Adding freq sample counts from v3 subsets...")
+    ht = annotate_v3_subsets_sample_count(ht)
+
+    logger.info("Removing 'Han' and 'Papuan' pops from freq and freq_meta...")
+    pops_to_remove = {"pop": ["han", "papuan"]}
+    ht = filter_freq_arrays(ht, pops_to_remove, keep=False, combine_operator="or")
+
+    logger.info("Updating HGDP pop labels and 'oth' -> 'remaining'...")
+    ht = update_pop_labels(ht, POP_MAP)
+
+    freq_hts = {"release": ht, **freq_hts}
+    for name, freq_ht in freq_hts.items():
+        logger.info("There are %i variants in the %s HT...", freq_ht.count(), name)
+        freq_hts[name] = freq_ht.select("freq").select_globals(*FREQ_GLOBALS)
+
+    logger.info("Joining release HT with AFs for added and subtracted samples...")
+    freq_hts = [freq_hts[x] for x in JOIN_FREQS]
+    ht = hl.Table.multi_way_zip_join(freq_hts, "ann_array", "global_array")
+
+    logger.info("Filtering joint HT to rows where at least one group has AC > 0...")
```

Comment:
Okay, so we filter here first?

Reply:
Yeah, I don't think there is any reason to store variants in this table that are not going to be in the release